proxy/src/ctx/http.rs

```diff
@@ -41,12 +47,19 @@ impl Request {
         client: &Arc<ctx::transport::Client>,
         id: usize,
     ) -> Arc<Self> {
+        // Look up whether the request has been extended with optional
+        // destination labels from the control plane's discovery API.
+        let dst_labels = request
+            .extensions()
+            .get::<discovery::DstLabels>()
```

Comment:
Aren't the labels per-connection instead of per-request? Why are the labels on the request, and not on `ctx::transport::Server` or `ctx::transport::Client`?

When we look up SO_ORIGINAL_DST, a per-connection value, we use:
```
            let orig_dst = req.extensions()
                .get::<Arc<ctx::transport::Server>>()
                .and_then(|ctx| {
                    ctx.orig_dst_if_not_local()
                });
```

This seems like the same sort of thing.

In particular, when the metadata for an IP address changes, that indicates that we're actually connecting to a different thing that happened to reuse a previously-in-use IP address, so we should drain the old connection, i.e. remove the old connection for the IP address out of the load balancer, and then add a new connection for the IP address to the load balancer, right?

(Since it's very late to change this now, I think we should file a follow-up issue to consider what we should do, if anything, when it looks like an IP address has been reused for a different destination, based on differing metadata.)

Reply:
I agree with both your comments and that it is rather late to change this now. We should continue discussing this.